etcd/etcd-live-cluster-reconfiguration.md

```diff
@@ -1,236 +1,350 @@
-Environment="ETCD_INITIAL_CLUSTER_STATE=existing"
-```
+  ```sh
+    $ etcdctl member add node2 http://0.0.0.2:2380
+    Added member named node2 with ID 4fb77509779cac99 to cluster
 
-**Note:** Make sure to remove the excess double quotes just after `ETCD_INITIAL_CLUSTER=` entry.
+    ETCD_NAME="node2"
+    ETCD_INITIAL_CLUSTER="52d2c433e31d54526cf3aa660304e8f1=http://0.0.0.1:2380,node2=http://0.0.0.2:2380,2cb7bb694606e5face87ee7a97041758=http://0.0.0.3:2380"
+    ETCD_INITIAL_CLUSTER_STATE="existing"
+    ```
 
-Parse the new drop-in:
+7. Modify the existing systemd drop-in, `/etc/systemd/system/etcd-member.service.d/20-clct-etcd-member.conf` by replacing the node data with the appropriate information from the output of the `etcdctl member add` command executed in the last step.
 
-```sh
-$ sudo systemctl daemon-reload
-```
+    ```ini
+    [Service]
+    ExecStart=
+    ExecStart=/usr/lib/coreos/etcd-wrapper $ETCD_OPTS \
+      --name="node2" \
+      --listen-peer-urls="http://0.0.0.0:2380" \
+      --listen-client-urls="https://0.0.0.2:2379,http://0.0.0.0:4001" \
+      --initial-advertise-peer-urls="http://<PEER_ADDRESS>:2380" \
+      --initial-cluster="52d2c433e31d54526cf3aa660304e8f1=http://0.0.0.1:2380,node2=http://0.0.0.2:2380,2cb7bb694606e5face87ee7a97041758=http://0.0.0.3:2380" \
+      --initial-cluster-state="existing" \
+      --initial-cluster-token="demo-etcd-token" \
+      --advertise-client-urls="http://<PEER_ADDRESS>:2379"
+    ```
 
-Check whether the new [drop-in][drop-in] is valid:
+8. Apply the changes to the cluster:
 
-```sh
-sudo journalctl _PID=1 -e -u etcd2
-```
+    `$ systemctl daemon-reload`
 
-And finally, if everything is ok start the `etcd2` service:
+9. Start the etcd member servie:
 
-```sh
-$ sudo systemctl start etcd2
-```
+  `$ sudo systemctl start etcd-member.service`
 
-Check cluster health:
+10. Check the cluster health:
 
-```sh
-$ etcdctl cluster-health
-```
+    ```sh
+    $ etcdctl cluster-health
+
+      member e6c2bda2aa1f2dcf is healthy: got healthy result from http://0.0.0.2:2379
+      cluster is healthy
+    ```
+
+If your cluster has healthy state, etcd successfully writes cluster configuration into the `/var/lib/etcd` directory.
+
+### Recovering etcd on CoreOS Container Linux
+
+#### etcd v3
+
+1. Download `etcdctl` from the [etcd Release page][etcd-release] and install, for example, into `/opt/bin`.
+
+2. Create a backup directory:
 
-If your cluster has healthy state, etcd successfully wrote cluster configuration into the `/var/lib/etcd2` directory. Now it is safe to remove the temporary `/run/systemd/system/etcd2.service.d/99-restore.conf` drop-in file.
+    `$ sudo mkdir /var/lib/etcd_backup`
 
-## etcd disaster recovery on CoreOS Container Linux
+3. Save the snapshot of the backup file:
+
+    `$ sudo ETCDCTL_API=3 /opt/bin/etcdctl snapshot save /var/lib/etcd_backup/backup.db`
+
+4. Restore the snapshot file:
```

Comment:
Restore the snapshot file into a new member directory `/var/lib/etcd_backup/etcd`: